### PR TITLE
feat: [SSCS-2]: SSCS MFE setup, base routes and module behind FF

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary
Creating the module at 75 instead of 85. We can plan for all modules to be at 75 unless there's a need.

image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-sscs/RouteDestinations.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import React from 'react'
+import { Redirect, Route, useParams } from 'react-router-dom'
+import type { ModulePathParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import type { SidebarContext } from '@common/navigation/SidebarProvider'
+import routes from '@common/RouteDefinitions'
+import { RouteWithLayout } from '@common/router'
+import { accountPathProps, projectPathProps } from '@common/utils/routeUtils'
+import { useAppStore } from 'framework/AppStore/AppStoreContext'
+import CIPipelineDeploymentList from '@ci/pages/pipeline-deployment-list/CIPipelineDeploymentList'
```
Comment (vkbansal-harness) [inline src/modules/85-sscs/RouteDestinations.tsx:16]: why use page from CI?

Comment (udayvunnam) [reply src/modules/85-sscs/RouteDestinations.tsx:16]: Right now We have CI, CD, STO, SSCS, IACM, and CF - all using the same code but duplicate components for the deployment list (STO already uses CIPipelineDeploymentList instead of copying the same). I will make a follow-up PR and make a single wrapper for all these and remove duplicate components

**Review by vkbansal-harness — DISMISSED** [review]

File: src/modules/10-common/RouteDefinitions.ts [hunk 2]
```diff
@@ -2015,7 +2015,20 @@ const routes = {
     }: PipelineType<ExecutionPathProps>) => {
       return `/iacm/orgs/${orgIdentifier}/projects/${projectIdentifier}/pipelines/${pipelineIdentifier}/${source}/${executionIdentifier}/resources`
     }
-  )
+  ),
+  // SSCS
+  toSSCS: withAccountId(() => '/sscs'),
+  toSSCSHome: withAccountId(() => '/sscs/home'),
+  toSSCSOverview: withAccountId(() => '/sscs/overview'),
+  toSSCSProjectOverview: withAccountId(
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/RouteDefinitions.ts:2023]: we already have a `toProjectOverview()` generic route which takes module param

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/75-sscs/components/SSCSSideNav.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
+  const { updateAppStore } = useAppStore()
+  const history = useHistory()
+
+  const showLinks = projectIdentifier && orgIdentifier
+  const params: ProjectPathProps & ModulePathParams & { accountId: string } = {
+    accountId,
+    projectIdentifier,
+    orgIdentifier,
+    module: 'sscs'
```
Comment (reetika04) [inline src/modules/75-sscs/components/SSCSSideNav.tsx:30]: Define module as a constant as it is reused everywhere.

File: src/modules/75-sscs/components/SSCSSideNav.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import ProjectSetupMenu from '@common/navigation/ProjectSetupMenu/ProjectSetupMenu'
+
+export default function SSCSSideNav(): React.ReactElement {
+  const { getString } = useStrings()
+  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
+  const { updateAppStore } = useAppStore()
+  const history = useHistory()
+
+  const showLinks = projectIdentifier && orgIdentifier
+  const params: ProjectPathProps & ModulePathParams & { accountId: string } = {
```
Comment (reetika04) [inline src/modules/75-sscs/components/SSCSSideNav.tsx:26]: { accountId: string } is not required. ProjectPathProps already contains accountId

File: src/modules/75-sscs/RouteDestinations.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      path={[
+        routes.toAllowDenyList({ ...accountPathProps, ...projectPathProps }),
+        routes.toSSCSOverview({ ...accountPathProps }),
+        routes.toProjectOverview({ ...accountPathProps, ...projectPathProps, module: 'sscs' })
+      ]}
+    >
+      <SSCSApp />
+    </RouteWithLayout>
+
+    <Route path="/account/:accountId/:module(sscs)">
```
Comment (reetika04) [inline src/modules/75-sscs/RouteDestinations.tsx:70]: Why is this needed?

File: src/modules/75-sscs/RouteDestinations.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      <RedirectToProjectOverviewPage />
+    </RouteWithLayout>
+
+    <RouteWithLayout
+      exact
+      sidebarProps={SSCSSideNavProps}
+      path={[
+        routes.toAllowDenyList({ ...accountPathProps, ...projectPathProps }),
+        routes.toSSCSOverview({ ...accountPathProps }),
+        routes.toProjectOverview({ ...accountPathProps, ...projectPathProps, module: 'sscs' })
```
Comment (reetika04) [inline src/modules/75-sscs/RouteDestinations.tsx:64]: You can pass moduleParams here instead of passing `module: 'sscs'`

File: src/modules/10-common/RouteDefinitions.ts [hunk 7]
```diff
@@ -2034,7 +2034,15 @@ const routes = {
...
       return `/iacm/orgs/${orgIdentifier}/projects/${projectIdentifier}/pipelines/${pipelineIdentifier}/${source}/${executionIdentifier}/resources`
     }
-  )
+  ),
+  // SSCS
+  toSSCS: withAccountId(() => '/sscs'),
+  toSSCSOverview: withAccountId(() => '/sscs/overview'),
+  toAllowDenyList: withAccountId(
+    ({ orgIdentifier, projectIdentifier }: Partial<ProjectPathProps>) =>
+      `/sscs/orgs/${orgIdentifier}/projects/${projectIdentifier}/allow-deny-list`
```
Comment (reetika04) [inline src/modules/10-common/RouteDefinitions.ts:2043]: Use getScopeBasedRoute

**Review by vkbansal-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]

File: src/modules/75-sscs/RouteDestinations.tsx (hunk 5 above)
Comment (udayvunnam) [reply src/modules/75-sscs/RouteDestinations.tsx:70]: We need pipelines and executions as well in sscs module

**Review by vkbansal-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]
